### PR TITLE
Cosmos state export and restart

### LINKED ISSUE #1183: Keep egress state in the Merkle tree
**What is the Problem Being Solved?**
In our current testnet, there is an ad-hoc way of exporting and reloading the egress addresses between incarnations of the chain, involving the "provisioning-server" and on-disk JSON databases.

**Description of the Design**

We have moved the provisioning state into the chain itself.  This started by creating a new `tx swingset provision-one` transaction.  We have also implemented a way to `query swingset egress`, and the Golang handlers to export and import this state so that it can be preserved between chains.

**Security Considerations**
The ability to provision a client should be controlled at the Golang layer by enforcing that the transaction sender account has at least `1provisionpass` token.

**Test Plan**

First, follow the steps to initialize a Dapp:

1. `agoric init foo`
2. `cd foo`
3. `agoric install`

Then, `(cd ~/agoric-sdk/packages/cosmic-swingset && make)`:

1. Create a new chain and provision a client address
  1. `agoric start local-chain --reset`
  2. `agoric start local-solo 8000 --reset`
  3. Test out the client at http://localhost:8000
2. Halt the client.
3. Halt the chain and export its state with `ag-chain-cosmos export --home=_agstate/agoric-servers/local-chain-26657 --for-zero-height >newgenesis.json`,
4. Validate that the provisioned egress address exists in the exported state
5. Reset the chain `ag-chain-cosmos unsafe-reset-all --home=_agstate/agoric-servers/local-chain-26657`
6. Copy the exported state to the reset chain `cp newgenesis.json _agstate/agoric-servers/local-chain-26657/config/genesis.json`
7. Restart the chain `agoric start local-chain`
9. Reset the client with `(cd _agstate/agoric-servers/local-solo-8000 && ~/agoric-sdk/packages/cosmic-swingset/bin/ag-solo reset-state)`
10. Start the client with `agoric start local-solo 8000` and test

### PR DESCRIPTION
**Pull request by michaelfig**
Closes #1183 

This implements the changes needed to smooth the path for the `halt; export; unsafe-reset-all; copy; start` process for restarting a local chain with existing coins, validators, and egress state, as described in #1183.

The next step along this path is to make the same process work for the installed testnet chain.

### CODE REVIEW

**Comment by michaelfig** [comment]
There is one introduced FIXME in `go.mod`.  I hope to remove the hack-around-cosmos lines next time we upgrade to a version of cosmos-sdk that fixes the export problem.

File: packages/cosmic-swingset/Makefile [hunk 1]
```diff
@@ -173,7 +173,7 @@ compile-gyp:
 install: go.sum
 	go install -v $(MOD_READONLY) $(BUILD_FLAGS) ./cmd/ag-cosmos-helper
 	bindir="$${GOBIN-$${GOPATH-$$HOME/go}/bin}"; \
-	  ln -sf "$$PWD/lib/ag-chain-cosmos" "$$PWD/bin/ag-nchainz" "$$bindir/"
+	  ln -sf "$$PWD/bin/ag-chain-cosmos" "$$PWD/bin/ag-nchainz" "$$bindir/"
```
Comment (warner) [inline packages/cosmic-swingset/Makefile:176]: I saw two other places that reference `.../lib/ag-chain-cosmos`: the cosmic-swingset README (`ln` instructions), and `scripts/chain-entry.sh`. Do those need to be updated too?

**Review by warner — APPROVED** [review]
LGTM

File: packages/cosmic-swingset/Makefile (hunk 1 above)
Comment (michaelfig) [reply packages/cosmic-swingset/Makefile:176]: Good catch!  I updated even more references now.

**Pull request merged**